### PR TITLE
add padding ratio limit to the context bucketing

### PR DESCRIPTION
**Pull request by yangulei**
- Generate prefix-prefill buckets with padding ratio limit for the context length.
- Implement the truncation and the `find_bucket` logic for prefix-prefill.

### CODE REVIEW

**Comment by yangulei** [comment]
@czhu15 @taotod @ranzhejiang 
This PR is ready for review now, please help to review, thanks!
I tested with sample code, and a complete testing and evaluation is necessary.
The padding ratio limit for context buckets reuses the `VLLM_PROMPT_SEQ_BUCKET_LIMIT`. And I recommend setting it to **0.1** as the sequence length and context length might be padded with the limit simultaneously.

File: vllm_hpu_extension/ops.py [hunk 1]
```diff
@@ -859,8 +859,10 @@ def fp8_channel_moe_prepare_weights(layer):
         _, _, hidden_size = layer.w13_weight.shape
         layer.moe_op.w13_weight = layer.w13_weight.reshape(-1, hidden_size).contiguous()
         layer.moe_op.w2_weight = layer.w2_weight.transpose(-1, -2).contiguous()
-        layer.moe_op.w13_weight_scale = torch.cat([layer.w13_weight_scale[i].unsqueeze(0) for i in range(layer.moe_op.num_experts)], dim = 0).reshape(-1)
```
Comment (czhu15) [inline vllm_hpu_extension/ops.py:862]: why these changes?

Comment (yangulei) [reply vllm_hpu_extension/ops.py:862]: Sorry, pushed wrong commit from another fix.

**Review by czhu15 — APPROVED** [review]
LGTM
